src/Stryker.Core/Stryker.Core.UnitTest/MutantFilters/IgnoredMethodMutantFilterTests.cs

```diff
@@ -14,13 +16,57 @@ namespace Stryker.Core.UnitTest.MutantFilters
 {
     public class IgnoredMethodMutantFilterTests : TestBase
     {
+
+
         [Fact]
         public static void ShouldHaveName()
         {
             var target = new IgnoredMethodMutantFilter() as IMutantFilter;
             target.DisplayName.ShouldBe("method filter");
         }
 
+        // filter should support documented cases
+        [Theory]
+        [InlineData("IgnoredMethod(true);", "true", true)]
+        [InlineData("x = IgnoredMethod(true);", null, true)]
+        [InlineData("var x = IgnoredMethod(true);", null, true)]
+        [InlineData("while (x == IgnoredMethod(true));", "==", false)]
+        [InlineData("IgnoredMethod()++;", "IgnoredMethod()++", false)]
+        [InlineData("x==1 ? IgnoredMethod(true): IgnoredMethod(false);", null, true)]
+        [InlineData("SomeMethod(true).IgnoredMethod(false);", "true", true)]
```

Comment:
I don't think this should be ignored. The true isn't part of the ignored method invocation

Reply:
Actually, the return value of `SomeMethod` is part of the invocation. What about this example:
`GetLogger(logAsync: true).LogMessage("logtext");` (named parameter for clarity). It will be hard to have discrimating test for this mutation: `GetLogger(logAsync: false).LogMessage("logtext");`

Of course, it does not mean in any way that `GetLogger` invocations are systematically ignored. Just when use as a chain invocation of `LogMessage`

Reply:
But we also mutate properties and other external values that could be used in `IgnoredMethod` right? I don't see that as part of the invocation of the ignored method

Reply:
It feels like you are over generalizing here. What this PR does is ignoring any mutations that solely impact the invocation of an ignored method in as an attempt to provide a more consistent and explainable experience.
It addresses known limitations with current logic, such as the ternary operator mutation `x==1 ? IgnoredMethod(true): IgnoredMethod(false);` while not addressing every one of them, such as :
```
if (loggingEnabled)
   Log(..);
```
where Stryker still mutations the `if` condition while it should not.

Reply:
I see your point of view. I'm fine with merging this now. If we see issues for this in the future we can always improve this further.